pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1786,6 +1786,12 @@ class Timedelta(_Timedelta):
                 return NaT
 
             reso = get_datetime64_unit(value)
+            if reso in [NPY_DATETIMEUNIT.NPY_FR_Y, NPY_DATETIMEUNIT.NPY_FR_M]:
```

Comment:
Based on my understanding no,  not without modifying `is_supported_unit()`. 

```python
is_supported_unit(NPY_DATETIMEUNIT reso):
    return (
        reso == NPY_DATETIMEUNIT.NPY_FR_ns
        or reso == NPY_DATETIMEUNIT.NPY_FR_us
        or reso == NPY_DATETIMEUNIT.NPY_FR_ms
        or reso == NPY_DATETIMEUNIT.NPY_FR_s
    )```
    
I think I'd need to add more units here. Also there would be a ?small performance hit as I understand things with going from `def` to `cpdef` on `is_supported_unit()` but probably not that big a deal.

Reply:
By using `not is_supported_unit` we will exclude Y, M and other units that we dont support

Reply:
Sorry I was under the impression that we still wanted to support `D` units and we just wanted to avoid ambitious units like `M` or `Y`.

If I change to `not is_supported_unit(reso)` then code like `pd.Timedelta(days=3).to_pytimedelta()` will now throw an error. If this is incorrect then I'll need to change approach.

Reply:
From the previous issue #52806 you mentioned: 
> should probably raise a ValueError. The equivalent pd.Timedelta(1, unit="M") now raises a ValueError

However `pd.Timedelta(1, unit="D")` still works as expected.

Reply:
We can use `is_supported_unit` but also allow `D`

Reply:
Proposed a slightly different approach based on the other patterns but unfortunately couldn't see how `is_supported_unit` works in this use case. Marking this as resolved in favour of the newer review.